### PR TITLE
tests: net: utils move to new ztest API

### PR DESCRIPTION
**Pull request by hunterhu3000**
Move net/utils tests to use new ztest API

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

### CODE REVIEW

**Review by rlubos — DISMISSED** [review]

**Review by enjiamai — DISMISSED** [review]

File: tests/net/utils/src/main.c [hunk 1]
```diff
@@ -402,7 +402,7 @@ static bool check_net_addr(struct net_addr_test_data *data)
 	return true;
 }
 
-void test_net_addr(void)
+ZTEST_USER(utils_fn_test_suite, test_net_addr)
```
Comment (nashif) [inline tests/net/utils/src/main.c:405]: no need for _test_suite here.

**Review by nashif — CHANGES_REQUESTED** [review]

File: tests/net/utils/src/main.c (hunk 1 above)
Comment (nashif) [reply tests/net/utils/src/main.c:405]: also, make sure you use net in the testsuite name, to make it networking specific. `utils_fn` can be anything.

Comment (hunterhu3000) [reply tests/net/utils/src/main.c]: Updated. Thanks to review

**Comment by hunterhu3000** [comment]
This test case can be run with CONFIG_ZTEST_SHUFFLE=y

**Review by enjiamai — DISMISSED** [review]

**Review by enjiamai — DISMISSED** [review]

**Review by rlubos — DISMISSED** [review]

**Comment by hunterhu3000** [comment]
@nashif Could you please help to review and +1, so that the PR can be merged in to the mainline? Thanks

**Review by stephanosio — DISMISSED** [review]

**Comment by hunterhu3000** [comment]
@nashif can we merge this PR?

**Comment by nashif** [comment]
done already.

**Pull request closed without merging**